code/_onclick/other_mobs.dm

```diff
@@ -42,10 +42,10 @@
 		if(istype(G) && G.Touch(A, 0)) // for magic gloves
 			return
 
-	if((LASER in mutations) && a_intent == INTENT_HARM)
+	if(HAS_TRAIT(src, TRAIT_LASEREYES) && a_intent == INTENT_HARM)
 		LaserEyes(A)
 
-	if(TK in mutations)
+	if(dna.GetSEState(GLOB.teleblock))
```

Comment:
Sure thing; so, this is partially prepping the way for TG's genetics update---they handle thing a bit more OOP than we do, so they have procs like `on_mutation_afterattack`, which TK, and a number of other mutations utilize, instead of having DNA checks all over the place like this; I opted to make it this way so it was easier to prep that for the future.